src/main/java/ch/njol/skript/sections/EffSecOpenInventory.java

```diff
@@ -0,0 +1,283 @@
+				"Opening a loom inventory requires Paper."),
+		SMITHING("smithing [table]", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a smithing table inventory requires Paper."),
+		STONECUTTER("stone[ ]cutter", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a stone cutter inventory requires Paper."),
+		WORKBENCH("(crafting [table]|workbench)");
+
+		private @Nullable String methodError;
+		private @Nullable Version version;
+		private final String property;
+		private boolean methodExists = true;
+
+		OpenableInventorySyntax(String property) {
+			this.property = property;
+		}
+
+		OpenableInventorySyntax(String property, Version version) {
+			this.property = property;
+			this.version = version;
+		}
+
+		OpenableInventorySyntax(String property, boolean methodExists, String methodError) {
+			this.methodExists = methodExists;
+			this.methodError = methodError;
+			this.property = property;
+		}
+
+		private String getFormatted() {
+			return this.toString().toLowerCase(Locale.ENGLISH) + ":" + property;
+		}
+
+		@Nullable
+		private Version getVersion() {
+			return version;
+		}
+
+		private boolean doesMethodExist() {
+			return methodExists;
+		}
+
+		@Nullable
+		private String getMethodError() {
+			return methodError;
+		}
+
+		private static String construct() {
+			StringBuilder builder = new StringBuilder("(");
+			OpenableInventorySyntax[] values = OpenableInventorySyntax.values();
+			for (int i = 0; i < values.length; i++ ) {
+				builder.append(values[i].getFormatted());
+				if (i + 1 < values.length)
+					builder.append("|");
+			}
+			return builder.append("|%-inventory%)").toString();
+		}
+	}
+
+	public static class InventorySectionEvent extends Event {
+
+		private final Inventory inventory;
+		private final Player[] players;
+
+		public InventorySectionEvent(Inventory inventory, Player... players) {
+			this.inventory = inventory;
+			this.players = players;
+		}
+
+		public Inventory getInventory() {
+			return inventory;
+		}
+
+		public Player[] getPlayers() {
+			return players;
+		}
+
+		@Override
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+
+	}
+
+	static {
+		EventValues.registerEventValue(InventorySectionEvent.class, Inventory.class, InventorySectionEvent::getInventory);
+		EventValues.registerEventValue(InventorySectionEvent.class, Player[].class, InventorySectionEvent::getPlayers);
+		Skript.registerSection(EffSecOpenInventory.class,
+				"(show|create|open) %inventory/inventorytype% (to|for) %players%",
+				"open [a[n]] " + OpenableInventorySyntax.construct() + " [view|window|inventory] (to|for) %players%",
+
+				"close [the] inventory [view] (of|for) %players%",
+				"close %players%'[s] inventory [view]");
+	}
+
+	private @Nullable OpenableInventorySyntax syntax;
+	private @Nullable Expression<?> object;
```

Comment:
```suggestion
	private @Nullable Expression<?> inventoryObject;
```
needs a more descriptive name